LibThreatClassic2.lua

```diff
@@ -1578,6 +1821,82 @@ function ThreatLib:IsActive()
 	return self.running or false
 end
 
+do
+	local targets = {}
+
+	_eventFrame:SetScript("OnUpdate", function (self, elapsed)
+		local time = GetTime()
+
+		if not ThreatLib.meleeRangeCheckInterval then
+			ThreatLib.meleeRangeCheckInterval = ThreatLib:GetMeleeRangeCheckInterval()
+		end
+
+		if time - lastMeleeRangeCheckTime > ThreatLib.meleeRangeCheckInterval then
+			local threatTargets_player = ThreatLib.threatTargets[playerGUID]
+
+			if threatTargets_player then
+				local current, currentTarget = next(targetIDs)
+
+				for targetGUID in pairs(threatTargets_player) do
+					local target = targets[targetGUID]
+
+					while not target and current do
+						local currentTargetGUID = UnitGUID(currentTarget)
+
+						if currentTargetGUID then
+							if currentTargetGUID == targetGUID then
+								target = currentTarget
+							else
+								targets[currentTargetGUID] = currentTarget
+							end
+						end
+
+						current, currentTarget = next(targetIDs, current)
+					end
+
+					local inMeleeRange
+
+					if target then
+						inMeleeRange = ThreatLib:UnitInMeleeRange(target)
+					end
+
+					ThreatLib:MeleeRangeUpdated(playerGUID, targetGUID, inMeleeRange)
+				end
+
+				wipe(targets)
+			end
+
+			local meleeRange_player = meleeRange[playerGUID]
+
+			if meleeRange_player then
```

Comment:
what's the purpose of this loop? Setting the meelerange to nil on all targets that have no threat?

Why?

Reply:
This ensures that the range entries of targets that have died are deleted. We could do this in `_clearThreat` and `_clearAllThreat` alternatively.

Reply:
How about adding it in another function and calling it here:
https://github.com/dfherr/LibThreatClassic2/blob/r6-alpha1/LibThreatClassic2.lua#L530-L533
that method could also use a rename (e.g. ScheduleCombatReset) or something.

Reply:
Also shouldn't this actually remove the `targetGUID` from the players meele range table and instead of setting it to nil.

Reply:
Removed. It's in `clearThreat`/`_clearAllThreat` now. These functions should renamed in the future to reflect that: their purpose is really to clear any kind of data after leaving combat.